stackdriver docs: metric query editor and annotations

daniellee opened the pull request:
Adds images and text for the query editor and annotations sections.

Note: changed all the image links to point to /img/docs/v53 instead of /img/docs/v54. On Grafana.org the images are in both directories until this is published, then I can remove /img/docs/v54.

daniellee committed:
stackdriver docs: metric query editor and annotations

marefr commented on docs/sources/index.md:
```diff
@@ -104,4 +104,12 @@ aliases = ["v1.1", "guides/reference/admin"]
       <img src="/img/docs/logos/icon_cloudwatch.svg">
       <h5>Cloudwatch</h5>
     </a>
+    <a href="{{< relref "features/datasources/mssql.md" >}}" class="nav-cards__item nav-cards__item--ds">
```
First, suggest to change the order:
1. Graphite
2. Elasticsearch
3. InfluxDB
4. Prometheus
5. Google Stackdriver
6. Cloudwatch
7. MySQL
8. Postgres
9. Microsoft SQL Server
10. OpenTSDB

Second, the CSS change of nav-cards and nav-cards__item doesn't work that good on mobile/tablet breakpoints for all cards on that page.

marefr replied:
On second though, how about having 5 items per row instead?

marefr commented on docs/sources/features/datasources/stackdriver.md:
```diff
@@ -43,44 +43,54 @@ To authenticate with the Stackdriver API, you need to create a Google Cloud Plat
-Choose a metric from the `Metric` dropdown.
+{{< docs-imagebox img="/img/docs/v53/stackdriver_query_editor.png" max-width= "400px" class="docs-image--right" >}}
+
+The Stackdriver query editor allows you to select metrics and group by multiple labels and use filter the results.
```
> ... and use filter the results.

Use what?

marefr reviewed (changes requested):
Minor change needed, see comment. 

One thing to consider would be to use a different syntax for images so that they'll get clickable - good for smaller screens. Referring to this format:

```
{{< docs-imagebox img="/img/docs/v53/annotation_tag_filter_variable.png" max-width="600px" >}}
```

daniellee committed:
docs: stackdriver fixes after review

- index page datasource order changed
- image links changed to the image box format
- incorrect sentence fixed

marefr reviewed (approved):
LGTM :+1:
